doc/7.stuff/keyboard.navigation/keyboard.patching-help.pd

```diff
@@ -0,0 +1,236 @@
+#N canvas 99 16 600 619 10;
+#X obj 57 323 print;
+#X text 296 132 Try it now \, starting on the trigger object.;
+#X obj 88 239 + 10;
+#X obj 118 268 select 5;
+#X obj 144 239 / 2;
+#X obj 240 239 * 15;
+#X msg 118 290 five;
+#N canvas 153 60 502 464 object_positioning 0;
+#X text 39 54 When you have an inlet/outlet selected and you create
+a new object/msg/float/ or symbol atom PD will position that new box
+automatically above/or below that object and connect it for you.;
+#X obj 41 128 notein;
+#X obj 271 397 list;
+#X obj 292 376 pack float float;
+#X obj 385 355 / 127;
+#X obj 385 334 r foo2;
+#X obj 292 355 - 1;
+#X obj 292 334 r foo1;
+#X obj 61 397 list;
+#X text 105 128 select the 3rd outlet and create an number atom.;
+#X text 37 272 Now you can effectively 'patch upwards'. Try selecting
+a inlet and creating objects. Select the [list] object on the left
+and recreate the object chain on the right., f 62;
+#X connect 3 0 2 1;
+#X connect 4 0 3 1;
+#X connect 5 0 4 0;
+#X connect 6 0 3 0;
+#X connect 7 0 6 0;
+#X restore 57 376 pd object_positioning;
+#N canvas 168 99 450 223 editing_objects 0;
+#X text 33 97 Try it:;
+#X obj 118 98 pack f f f;
+#X text 33 37 You can now enter/exit the object's text editing but
+using mod+return.;
+#X restore 57 403 pd editing_objects;
+#N canvas 194 97 435 301 the_canvas_message_console 0;
+#X text 32 64 It allows to send messages to the current patch window.
+;
+#X text 30 147 This is the same as the following:;
+#X obj 43 211 s \$0-thisCanvas;
+#X obj 42 248 namecanvas \$0-thisCanvas;
+#X text 32 26 Press Ctrl + ' (single quote) to open the canvas message
```

Comment:
see my comment below (in `pd_bindings.tcl`): many keyboard layouts will not have doublequote on <kbd>Shift</kbd>+<kbd>'</kbd>!

Reply:
True! 

I was thinking maybe <kbd>Ctrl/Cmd</kbd> + <kbd>Shift</kbd> + <kbd>some letter</kbd> would be better. Do you have a suggestion?

Reply:
well, you could also just document that one should press `"` (double-quote) to get that behaviour, as this is what you are binding to.

alternatively, you could re-use the <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>m</kbd> shortcut, for sending messages:
- when on the Pd-console window, this sends a generic message into Pd-land
- when on a normal canvas, this could send a message to the canvas itself.

but on re-thinking this, i actually think that this feature should be included at all.
i think the audience is pretty small, and the terminal (which will mostly generate messages like `canvas: no method for 'foo'`) will be confusing to most people.

you can easily provide it as a gui-plugin, so power-users can selectively install it.


(i guess the main reason to include it is `goto`: instead you could <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>o</kbd> open a specialised "goto" bar, that might even have a number dial, to limit the possibilities to valid indices)

Reply:
I agree with you and I'm inclined to go with the "not include the canvas console at all and release as gui-plugin" route. Maybe somebody may comment on the issue but if that doesn't happen I'll remove it in favor of the dedicated embedded dialogs for selecting and connecting.